contexts/BalancesContext.tsx

```diff
@@ -71,6 +73,46 @@ export function BalancesProvider({
     [connection, owner],
   );
 
+  async function findTokenAccountsByOwner(walletPublicKey: PublicKey): Promise<PublicKey[]> {
+    const { value } = await connection.getTokenAccountsByOwner(walletPublicKey, {
+      programId: new PublicKey(TOKEN_PROGRAM_ID),
+    });
+
+    return value.map(({ pubkey }) => pubkey);
+  }
+
+  async function subscribeToTokenBalances() {
+    if (!owner) return;
+    const tokenAccountPubkeys = await findTokenAccountsByOwner(owner);
+
+    for (const pubkey of tokenAccountPubkeys) {
+      connection.onAccountChange(pubkey, (accountInfo: AccountInfo<Buffer>) => {
+        const accountData = AccountLayout.decode(accountInfo.data);
+
+        const dividedTokenAmount = new BN(accountData.amount) / new BN(LAMPORTS_PER_SOL);
+        const tokenVal: TokenAmount = {
+          amount: dividedTokenAmount.toString(),
+          decimals: 9,
```

Comment:
So this works for META, but not USDC as USDC has 6 decimals.

Reply:
Okay addressed this with a fix.